### DISCUSSION TITLE
Bring structure to blazor

### DISCUSSION THREAD

**Discussion by ghost** [pamaxie/--OLD--Pamaxie#36] [General]
Currently developing in Blazor for more complex things is an absolute mess. We lack design patterns. Please we need either someone to come up with one that works for us (and maybe also other developers) or use one that already exists.

Discussion is open and and recommendations on existing design patterns are welcome. I know this is a bit to ask but it should be flexible enough to develop websites with potentially hundreds of pages without loosing an overview of what we want to achieve.

**Comment by ghost** [comment]
So from what I can see there are no advertised design patterns by microsoft for server side blazor. For client side MVC seems to be the preferred model, but since server and client side blazor work differently we may have to make one up.
Currently I would love to have something like MVVM where we seperate the View from the Model. I'm not sure on how practical an implementation like that would be though.

**Comment by PKoldborg** [comment]
I made a comit regarding our structure for our website.
[#36 Initial structuralisation of Blazor website](https://github.com/pamaxie/Pamaxie/commit/8ab0a3b7ab7d78d38291a0748faafc2381df1a9f)
